mlflow/utils/autologging_utils.py

```diff
@@ -9,3 +11,62 @@ def try_mlflow_log(fn, *args, **kwargs):
         fn(*args, **kwargs)
     except Exception as e:  # pylint: disable=broad-except
         warnings.warn("Logging to MLflow failed: " + str(e), stacklevel=2)
+
+
+def param_logger(fn, args=[], kwargs=None, unlogged=[]):
+    """
+    Log parameters explicitly passed to a function.
+    :param fn: function whose parameters are to be logged
+    :param args: arguments explicitly passed into fn
+    :param kwargs: kwargs explicitly passed into fn
+    :param unlogged: parameters not to be logged
+    :return: None
+    """
+    # Names of all parameters for the function
+    all_param_names = inspect.getargspec(fn)[0]
+
+    # Default values of all parameters with default values. Has length of n, and corresponds
+    # to values of last n elements in all_param_names
+    all_default_values = inspect.getargspec(fn)[3]
+
+    # Checking if default values are present for logging. Known bug that getargspec will return an
+    # empty argspec for certain functions, despite the functions having an argspec.
+    if all_default_values:
```

Comment:
Maybe we should check `len(all_default_values) > 0` for clarity? Although it's interesting that the truth value of an empty list is False